app/models/measure_condition_permutations/calculator.rb

```diff
@@ -0,0 +1,110 @@
+module MeasureConditionPermutations
+  class Calculator
+    # There are 2 rules for calculating permutations
+    #
+    # When there are conditions with matching permutation_keys, then a single
+    # permutation group is provided with all permutations
+    #
+    # If there are no conditions with matching permutation key, then generate
+    # one group per condition code, with a separate permutation per condition
+    # within it
+
+    def initialize(measure)
+      @measure = measure
+    end
+
+    def measure_conditions
+      @measure_conditions = @measure.measure_conditions
```

Comment:
did you mean to memoize here? `||=`

Reply:
also, where is this method used? I saw it only in the tests,
maybe I am missing something

Reply:
Hmm - good question, I think I decided against memoizing it but that means it doesn't need assigning to a variable - code moved around a lot, I think this was moved from the initializer where direct assignment would make sense.

I'll revisit this

Reply:
I think this is used in the serializer?